File: tsp-solver-frontend/src/app/model/request/tsp/GeneticAlgorithmRequest.ts
import {CrossoverType} from "./ga/CrossoverType";
import {SelectionType} from "./ga/SelectionType";

export class GeneticAlgorithmRequest {
  costMatrix: number[][]
  populationSize: number;
  elitismSize: number;
  mutationRate: number;
  epochsNumber: number
  maxEpochsNoImprovement: number
  selectionType: SelectionType
  crossoverType: CrossoverType

  constructor(costMatrix: number[][], populationSize: number, elitismSize: number, mutationRate: number,
              epochsNumber: number, maxEpochsNoImprovement: number, selectionType: SelectionType, crossoverType: CrossoverType) {
    this.costMatrix = costMatrix;
    this.populationSize = populationSize;
    this.elitismSize = elitismSize;
    this.mutationRate = mutationRate;
    this.epochsNumber = epochsNumber;
    this.maxEpochsNoImprovement = maxEpochsNoImprovement;
    this.selectionType = selectionType;
    this.crossoverType = crossoverType;
  }

}

File: tsp-solver-backend/src/main/java/pl/edu/pbs/request/tsp/GeneticAlgorithmRequest.java
package pl.edu.pbs.request.tsp;

import lombok.Data;
import pl.edu.pbs.tsp.algorithm.ga.CrossoverType;
import pl.edu.pbs.tsp.algorithm.ga.SelectionType;

@Data
public class GeneticAlgorithmRequest {

    private double[][] costMatrix;
    private int populationSize;
    private int elitismSize;
    private double mutationRate;
    private int epochsNumber;
    private int maxEpochsNoImprovement;
    private SelectionType selectionType;
    private CrossoverType crossoverType;

}
